### PR TITLE
Remove jslib-api private dependency

### PR DESCRIPTION
**Pull request by nandito**
* add storybook's static folder to `.gitignore`
* move the necessary api functions to this lib, so we can remove the jslib-api package
* migrate jslib-api's sinon+mocha+chai tests to Jest

## Test plan

1. Install dependencies and run the tests locally: `yarn test`
2. Start the storybook and make sure the stories run as expected

### CODE REVIEW

File: src/lib/api/ApiClient.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    constructor({ httpClient, fetchDeviceCredentials }: AuthenticatedHttpClientOptions) {
+        this._httpClient = httpClient;
+        this._fetchDeviceCredentials = fetchDeviceCredentials;
+    }
+
+    request(url: string, options: HttpClientRequestConfig): Promise<Response> {
+        return this._fetchDeviceCredentials().then((credentials) => {
+            const headers = Object.assign({}, options.headers, _getAuthHeader(credentials), {
+                "X-Appearin-Device-Platform": "web",
```
Comment (havfo) [inline src/lib/api/ApiClient.ts:42]: Hmm...this is going to be a public SDK, is it ok to still have `Appearin` here?

File: src/lib/api/ApiClient.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ */
+export default class ApiClient {
+    authenticatedHttpClient: AuthenticatedHttpClient;
+    authenticatedFormDataHttpClient: MultipartHttpClient;
+
+    /**
+     * Create an ApiClient instance.
+     */
+    constructor({
+        baseUrl = "https://api.appearin.net",
```
Comment (havfo) [inline src/lib/api/ApiClient.ts:67]: Could we get in trouble for still using this, in an SDK?

File: src/lib/api/Credentials.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        };
+        this.hmac = hmac;
+        this.userId = userId;
+    }
+
+    toJson(): Json {
+        return {
+            credentials: this.credentials,
+            hmac: this.hmac,
+            ...(this.userId && { userId: this.userId }),
```
Comment (havfo) [inline src/lib/api/Credentials.ts:34]: `this.userId` is either a value or `undefined`, so it should be fine to do:
```
{
            credentials: this.credentials,
            hmac: this.hmac,
            userId: this.userId,
}

File: src/lib/api/MultipartHttpClient.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    /**
+     * Convert the provided object to a FormData object containing the same keys and values.
+     * @param {object} data - the data to convert.
+     * @returns {FormData}
+     */
+    static dataToFormData(data: Record<string, string>): FormData {
+        assert.ok(data, "data is required");
+
+        const fd = new FormData();
+        Object.keys(data).forEach((key) => {
```
Comment (havfo) [inline src/lib/api/MultipartHttpClient.ts:26]: A more straight forward way would be to do:
```
for (const [ key, value ] of Object.entries(data)) {
    fd.append(key, value);
}
```

File: src/lib/api/Response.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+export default class Response {
+    data: Json;
+    headers: Record<string, unknown>;
+    status: number;
+    statusText: string;
+    url: string | null;
+
+    constructor(initialValues: ResponseOptions = {}) {
+        this.data = initialValues.data === undefined ? {} : initialValues.data;
```
Comment (havfo) [inline src/lib/api/Response.ts:28]: It would be appropriate with the nullish coalescing operator here I think `this.data = initialValues.data ?? {}`

**Review by havardholvik — APPROVED** [review]
Works as expected, good move 👍 

I guess these TS warnings also existed in jslib-api
``` 
/Users/havardholvik/Development/Whereby/browser-sdk/src/lib/api/HttpClient.ts
  8:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/Users/havardholvik/Development/Whereby/browser-sdk/src/lib/api/Response.ts
   6:30  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  14:30  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/Users/havardholvik/Development/Whereby/browser-sdk/src/lib/api/modules/tests/ChromeStorageStore.spec.ts
  21:68  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  32:68  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  45:56  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  45:65  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  58:56  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  58:65  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/Users/havardholvik/Development/Whereby/browser-sdk/src/lib/api/parameterAssertUtils.ts
  54:69  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/Users/havardholvik/Development/Whereby/browser-sdk/src/lib/api/test/helpers.ts
  19:64  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  28:71  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

✖ 12 problems (0 errors, 12 warnings)

```

**Pull request merged**